models/action.go

```diff
@@ -672,33 +672,37 @@ func MergePullRequestAction(actUser *User, repo *Repository, pull *Issue) error
 	return mergePullRequestAction(x, actUser, repo, pull)
 }
 
-// GetFeeds returns action list of given user in given context.
-// actorID is the user who's requesting, ctxUserID is the user/org that is requested.
-// actorID can be -1 when isProfile is true or to skip the permission check.
-func GetFeeds(ctxUser *User, actorID, offset int64, isProfile bool) ([]*Action, error) {
+// GetFeedsOptions options for retrieving feeds
+type GetFeedsOptions struct {
+	RequestedUser    *User
+	RequestingUserID int64
+	IncludePrivate   bool // include private actions
+	OnlyPerformedBy  bool // only actions performed by requested user
+}
+
+// GetFeeds returns actions according to the provided options
+func GetFeeds(options GetFeedsOptions) ([]*Action, error) {
 	actions := make([]*Action, 0, 20)
-	sess := x.
-		Limit(20, int(offset)).
+	sess := x.Limit(20).
 		Desc("id").
-		Where("user_id = ?", ctxUser.ID)
-	if isProfile {
-		sess.
-			And("is_private = ?", false).
-			And("act_user_id = ?", ctxUser.ID)
-	} else if actorID != -1 && ctxUser.IsOrganization() {
-		env, err := ctxUser.AccessibleReposEnv(actorID)
+		Where("user_id = ?", options.RequestedUser.ID)
+	if options.OnlyPerformedBy {
+		sess.And("act_user_id = ?", options.RequestedUser.ID)
+	}
+	if !options.IncludePrivate {
+		sess.And("is_private = ?", false)
+	}
+	if options.RequestedUser.IsOrganization() {
+		env, err := options.RequestedUser.AccessibleReposEnv(options.RequestingUserID)
 		if err != nil {
 			return nil, fmt.Errorf("AccessibleReposEnv: %v", err)
 		}
-		repoIDs, err := env.RepoIDs(1, ctxUser.NumRepos)
+		repoIDs, err := env.RepoIDs(1, options.RequestedUser.NumRepos)
```

Comment:
I don't see a nice way to use the same connection in both places (i.e. the `sess` variable and this line). If you have a more specific suggestion about how to only make one connection, let me know. 

For what it's worth, the first session (i.e. the `sess` variable) won't actually connect to the database until line 704, so I don't think the two connections will ever overlap.